python/ray/actor.py

```
@@ -832,58 +727,142 @@ def __ray_checkpoint__(self):
+        checkpoint id to persistent storage.
+
+        Args:
+            actor_id: Actor’s ID.
+            checkpoint_id: ID of this checkpoint. You should save it together
+                with actor’s checkpoint data. And it will be used by the
+                `load_checkpoint` method.
+        Returns:
+            None.
+        """
+        pass
+
+    @abstractmethod
+    def load_checkpoint(self, actor_id, available_checkpoints):
+        """Load actor's previous checkpoint, and restore actor's state.
+
+        This method will be called when an actor is reconstructed, after
+        actor's constructor.
+        If the actor needs to restore from previous checkpoint, this function
+        should restore actor's state and return the checkpoint ID. Otherwise,
+        it should do nothing and return None.
+
+        Args:
+            actor_id: Actor’s ID.
+            available_checkpoints: A list of `Checkpoint` namedtuples that
+            contains all available checkpoint IDs and their timestamps, sorted
+            by timestamp in descending order. Note, this method must return the
+            ID of one checkpoint in this list, or None. Otherwise, an exception
+            will be raised.
+        Returns:
+            The ID of the checkpoint from which the actor was resumed, or None
+            if the actor should restart from the beginning.
+        """
+        pass
+
+    @abstractmethod
+    def checkpoint_expired(self, actor_id, checkpoint_id):
+        """Delete an expired checkpoint.
+
+        This method will be called when an checkpoint is expired. You should
+        implement this method to delete your application checkpoint data.
+        Note, the maximum number of checkpoints kept in the backend can be
+        configured at `RayConfig.num_actor_checkpoints_to_keep`.
+
+        Args:
+            actor_id: ID of the actor.
+            checkpoint_id: ID of the checkpoint that has expired.
+        Returns:
+            None.
```

Maybe also return `True`/`False`.

---

Do you want the return value to indicate whether the application checkpoint was deleted? I think the backend doesn't care about this.

---

Ah, then maybe document that the user should throw an exception if the checkpoint wasn't deleted. From a user perspective, it wasn't clear to me how I should handle failing to delete a checkpoint.